_posts/2019-01-16-dask-extension-arrays.md

```diff
@@ -0,0 +1,210 @@
+                                       dtype=dtype)
+```
+
+Now users of that extension type can place those arrays inside a Dask DataFrame
+or Series.
+
+
+```python
+>>> df = pd.DataFrame({"A": DecimalArray([Decimal('1.0'), Decimal('2.0'),
+...                                       Decimal('3.0')])})
+
+>>> ddf = dd.from_pandas(df, 2)
+>>> ddf
+Dask DataFrame Structure:
+                     A
+npartitions=1
+0              decimal
+2                  ...
+Dask Name: from_pandas, 1 tasks
+
+>>> ddf.dtypes
+A    decimal
+dtype: object
+```
+
+And from there, the usual operations just as the would in pandas.
+
+```python
+>>> from random import choices
+>>> df = pd.DataFrame({"A": DecimalArray(choices([Decimal('1.0'),
+...                                               Decimal('2.0')],
+...                                              k=100)),
+...                    "B": np.random.choice([0, 1, 2, 3], size=(100,))})
+>>> ddf = dd.from_pandas(df, 2)
+In [35]: ddf.groupby("A").B.mean().compute()
+Out[35]:
+A
+1.0    1.50
+2.0    1.48
+Name: B, dtype: float64
+
+```
+
+The Real Lesson
+---------------
+
+It's neat that Dask now supports extension arrays. But to me, the exciting thing
+is just how little work this took. The [PR implementing
+this](https://github.com/dask/dask/pull/4379/files) is quite short, just
+defining the object that third-parties register with, and using it to generate
+the data when dtype is detected. And for third-party extension array authors,
+the work is minimal.
+
+This highlights the importance of one of the Dask project's core values: working
+with the community. If you visit [dask.org](https://dask.org), you'll see
+phrases like
+
+> Integrates with existing projects
+
+and
+
+> Built with the broader community
+
+At the start of Dask, the developers *could* have gone off an re-written pandas
+or NumPy from scratch to be parallel friendly (though we'd probably still be
+working on that part today, since that's such a massive undertaking). Instead,
+the Dask developers worked with the community, occasionally nudging it in
+directions that would help out dask. For example, many places in pandas [held
+the GIL](http://matthewrocklin.com/blog/work/2015/03/10/PyData-GIL), preventing
+thread-based parallelism. Rather than abandoning pandas, the Dask and pandas
+developers worked together to release the GIL where possible when it was a
+bottleneck for `dask.dataframe`. This benefited Dask and anyone else trying to
+do thread-based parallelism with pandas DataFrames.
+
+And now, when pandas introduces new features like nullable integers,
+`dask.dataframe` just needs to register it as an extension type and immediately
+benefits from it. And third-party extension array authors can do the same for
+their extension arrays.
+
+If you're writing an ExtensionArray, make sure to add it to the [pandas
+ecosystem](http://pandas.pydata.org/pandas-docs/version/0.24/ecosystem.html#extension-data-types)
+page, and register it with Dask.
+
```

Comment:
```suggestion

If you're looking to contribute on a fun project, we recommend trying to connect Dask dataframe up to an existing extension array like [cyberpandas](https://github.com/ContinuumIO/cyberpandas) or TODO.  

```

Reply:
Also, should we mention these sorts of projects at the beginning to add to the motivation?  Currently all of the new functionality we present is decimal.